src/app/shared/models/teacher.model.ts

```diff
@@ -1,4 +1,10 @@
-export class Teacher {
+export interface Person {
+  id?: string;
+  lastName: string;
+  middleName?: string;
+  firstName: string;
+}
```

Comment:
noooo
There should be just one interface, that is extended by teacher, child, etc